Lib/urllib/parse.py

```diff
@@ -893,10 +893,12 @@ def quote_from_bytes(bs, safe='/'):
     not perform string-to-bytes encoding.  It always returns an ASCII string.
     quote_from_bytes(b'abc def\x3f') -> 'abc%20def%3f'
     """
-    if not isinstance(bs, (bytes, bytearray)):
-        raise TypeError("quote_from_bytes() expected bytes")
     if not bs:
         return ''
+
+    if not isinstance(bs, (bytes, bytearray)):
+        raise TypeError("quote_from_bytes() expected bytes")
+    
```

Comment:
Thank you for your contribution. 

This changes behavior, and if it didn't break any of the unit tests it means this case is not covered by tests and you need to fix that in this PR. 

You also need to create an issue on bugs.python.org to describe this change and confirm that this is a bug and not a feature.

Reply:
I would see how the check `not bs` is redundant though, my approval was based on this.. but I agree...confirm functionality and add tests.

Reply:
Ah, it's not redundant:

>>> isinstance(None, (bytes, bytearray))
False

so before the change None gave TypeError and after the change it returns ''.

Reply:
I think thats what the logic should be, None == `not bs` hence '', Type Error is meant for the instance check, no?

Reply:
I don't follow. The change is this:

```
def old():
    bs = None
    if not isinstance(bs, (bytes, bytearray)):
        raise TypeError("quote_from_bytes() expected bytes")
    if not bs:
        return ''

def new():
    bs = None
    if not bs:
        return ''
    if not isinstance(bs, (bytes, bytearray)):
        raise TypeError("quote_from_bytes() expected bytes")

print('--------- new:')
print(new() == '')
print('--------- old:')
print(old() == '')
```

output: 
```
--------- new:
True
--------- old:
Traceback (most recent call last):
  File "C:\Users\User\src\cpython-current\tmp.py", line 20, in <module>
    print(old() == '')
  File "C:\Users\User\src\cpython-current\tmp.py", line 6, in old
    raise TypeError("quote_from_bytes() expected bytes")
TypeError: quote_from_bytes() expected bytes
```

Reply:
I mean  `not None` == True hence the correct output should '' , even for the old logic, no? that's what am saying. But it also passes for the TypeError case, so making the instance check come first makes the `not bs` redundant for this case.